### ISSUE TITLE
BUG WITH SOLUTION: GetDeviceMemInfo Cast Incorrect

### ISSUE THREAD

**Issue by mbojrab** [Theano/Theano#5603]
Using Theano on Win10 x64 with GTX1070, Cuda 8.0, VC++ 2013 Community.

Was getting negative values from:
```python
import theano.sandbox.cuda.basic_ops as sbcuda
sbcuda.cuda_ndarray.cuda_ndarray.mem_info()
```

I installed new drivers, got the latest version of theano. Finally put together a test program in VC++ (taking Theano out of equation).

```C++
#include <iostream>
#include <cuda_runtime.h>

int main()
{
	size_t free = 0;
	size_t total = 0;
	cudaMemGetInfo(&free, &total);

	std::cout << "free : " << free << std::endl;
	std::cout << "total: " << total << std::endl;

	std::cout << long(free) << std::endl;
	std::cout << long(total) << std::endl;

	std::cout << sizeof(size_t) << std::endl;
	std::cout << sizeof(long) << std::endl;

	return 0;
}
```
Output:
```
free : 7139626188
total: 8589934592
-1450308404
0
8
4
Press any key to continue . . .
```

I tracked the bug down to cuda_ndarray.cu, and I'm attaching a potential fix to the bug. This will manifest in a compiler specific way. For cl.exe `sizeof(long) == 4`, so it presents a problem. I've implemented the fix in "develop" mode, and it works on my end. I don't have a means currently to test other platforms.

The text file below is a diff patch for my proposed updates:
[cuda_ndarray.txt](https://github.com/Theano/Theano/files/794705/cuda_ndarray.txt)

Thank you!

**Comment by nouiz** [comment]
The fix look good. If you make a PR for it, I'll merge it. We can test it
on linux easily.

But just to note will shorlty after Theano 0.9 release delete the old
back-end code. So the fix won't stay for long in Theano.

So if you won't make a PR fix the fix, I'll just close this issue.

thanks

**Comment by mbojrab** [comment]
I saw this was going away. I attempted to build/install the new libgpuarray backend, but hadn't gotten this working yet on Windows. I'll keep working at it, and may suggest an update to the documentation.

For the PR, do you accept these through forking? I was unable to create a branch in the main repo, so if I fork out, make the mod and PR the update, would that work?

**Comment by mbojrab** [comment]
nvm, found my answer in the Developer Start Guide.

**Comment by lamblin** [comment]
Fixed by #5605.
